### PR TITLE
Update BLAST to 2.16.0

### PR DESCRIPTION
**Pull request by ebete**
Updates BLAST from 2.15.0 to 2.16.0.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by corneliusroemer** [comment]
`latest` seems to still point at 2.15, and docs don't mention 2.16 at all - how did you get it :D 

See:
```
wget https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/   
```

**Comment by corneliusroemer** [comment]
@ebete I've noticed you've effectively maintained this recipe but are missing from the `maintainers` section in the recipe. Should I add you? :)

Google Chrome 2024-07-17 12 52 27

**Comment by ebete** [comment]
@corneliusroemer Yeah I stumbled onto this release by chance :)

I'm fine with being added in the `recipe-maintainers` section. I'm not a BLAST+ dev though, if that matters.

**Comment by corneliusroemer** [comment]
@ebete Great! The recipe entry does not mean that you maintain the _tool_ but the _bioconda recipe_ - it's helpful to know who to ping :)

**Comment by corneliusroemer** [comment]
While we're at it - we could try to build for osx-arm64 and linux-aarch64 as well, maybe once this new version works for linux, new build number

**Comment by martin-g** [comment]
> While we're at it - we could try to build for osx-arm64 and linux-aarch64 as well

See https://github.com/bioconda/bioconda-recipes/pull/46237

**Comment by ebete** [comment]
Missing `dlfcn.h`, [if I'm deciphering this correctly](https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=64165&view=logs&j=1b052f1d-4456-52f0-9d43-71c4c5bd734d&t=34540139-28f3-5244-2e84-5ff07800d4d9&l=216)?

**Comment by ebete** [comment]
@BiocondaBot please fetch artifacts

**Comment by ebete** [comment]
I'm not really sure what's going wrong in the osx-64 build, as it builds fine locally using bioconda-utils and the error is not that clear. So I skipped it for now so the linux64 build can be published at least.
I'll try to look into it further later.

**Comment by ebete** [comment]
@BiocondaBot please add label

**Comment by epruesse** [comment]
I suppose you could take me out as maintainer... it's been years. ;)

**Review by epruesse — APPROVED** [review]

**Comment by epruesse** [comment]
I've approved this. However, I know how these things go. If you skip OSX now, it will remain skipped until someone needs it desperately enough to re-enable it. Any chance you could have another look at that?

**Comment by ebete** [comment]
The plan after this is to get the osx-arm64 build working, and hopefully get the normal amd64 build working at the same time. The issue for me is that I don't have an Intel Mac anymore, meaning local debugging build problems for that arch is difficult.

**Comment by epruesse** [comment]
Ok. I'm going to merge this. Thank you!

File: recipes/blast/meta.yaml [hunk 1]
```diff
@@ -25,7 +25,8 @@ source:
       - normbase.patch
 
 build:
-  number: 1
+  skip: True # [osx]
```
Comment (corneliusroemer) [inline recipes/blast/meta.yaml:28]: Would be nice next time to add a link to the PR/issue describing why it fails :)

**Pull request merged**